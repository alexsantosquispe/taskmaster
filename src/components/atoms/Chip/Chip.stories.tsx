import type { Meta, StoryObj } from '@storybook/react';
import { Chip } from './Chip';

const meta = {
  title: 'Atoms/Chip',
  component: Chip,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof Chip>;

export default meta;

type Story = StoryObj<typeof Chip>;

export const Default: Story = () => {
  return <Chip label="Default Chip" />;
};

Default.args = {};
