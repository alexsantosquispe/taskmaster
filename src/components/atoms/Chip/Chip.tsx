import { twMerge } from 'tailwind-merge';

interface ChipProps {
  label: string;
  className?: string;
}

export const Chip = ({ label, className }: ChipProps) => {
  return (
    <div
      className={twMerge(
        'text-primary flex w-fit min-w-13 items-center justify-center gap-2 rounded-xl bg-gray-200 px-2 py-0.5 text-xs font-medium dark:bg-neutral-700 dark:text-white',
        className
      )}
    >
      <span>{label}</span>
    </div>
  );
};
