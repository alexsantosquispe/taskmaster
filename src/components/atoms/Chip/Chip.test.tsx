import { render } from '@testing-library/react';
import { Chip } from './Chip';

describe('Chip', () => {
  it('should render the component correctly', () => {
    const component = render(<Chip label="test" />);

    expect(component).toMatchSnapshot();
  });
});
