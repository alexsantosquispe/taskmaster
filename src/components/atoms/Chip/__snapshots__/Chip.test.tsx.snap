// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Chip should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-primary flex w-fit min-w-13 items-center justify-center gap-2 rounded-xl bg-gray-200 px-2 py-0.5 text-xs font-medium dark:bg-neutral-700 dark:text-white"
      >
        <span>
          test
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-primary flex w-fit min-w-13 items-center justify-center gap-2 rounded-xl bg-gray-200 px-2 py-0.5 text-xs font-medium dark:bg-neutral-700 dark:text-white"
    >
      <span>
        test
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
